examples/view-integration/external-views/src/app.ts

```diff
@@ -18,59 +23,53 @@ import { createElement } from "react";
 // eslint-disable-next-line import/no-internal-modules
 import { createRoot } from "react-dom/client";
 
-import { DiceRollerContainerRuntimeFactory } from "./containerCode.js";
-import type { IDiceRoller } from "./interface.js";
+import { DiceRollerContainerRuntimeFactory, type IDiceRoller } from "./container/index.js";
 import { DiceRollerView } from "./view.js";
 
-const updateTabForId = (id: string): void => {
-	// Update the URL with the actual ID
-	location.hash = id;
-
-	// Put the ID in the tab title
-	document.title = id;
-};
-
-const render = (diceRoller: IDiceRoller): void => {
-	const appDiv = document.getElementById("app") as HTMLDivElement;
-	const appRoot = createRoot(appDiv);
-	appRoot.render(createElement(DiceRollerView, { diceRoller }));
-};
-
 const urlResolver = createInsecureTinyliciousTestUrlResolver();
 const tokenProvider = createInsecureTinyliciousTestTokenProvider();
 const documentServiceFactory = createRouterliciousDocumentServiceFactory(tokenProvider);
-const codeLoader = new StaticCodeLoader(new DiceRollerContainerRuntimeFactory());
+const codeLoader: ICodeDetailsLoader = {
+	load: async (details: IFluidCodeDetails): Promise<IFluidModuleWithDetails> => {
+		return {
+			module: { fluidExport: new DiceRollerContainerRuntimeFactory() },
+			details,
+		};
+	},
+};
 
-async function start(): Promise<void> {
-	let id: string;
-	let diceRoller: IDiceRoller;
+let id: string;
+let container: IContainer;
 
-	if (location.hash.length === 0) {
-		const container = await createDetachedContainer({
-			codeDetails: { package: "1.0" },
-			urlResolver,
-			documentServiceFactory,
-			codeLoader,
-		});
-		diceRoller = (await container.getEntryPoint()) as IDiceRoller;
-		await container.attach(createTinyliciousTestCreateNewRequest());
-		if (container.resolvedUrl === undefined) {
-			throw new Error("Resolved Url not available on attached container");
-		}
-		id = container.resolvedUrl.id;
-	} else {
-		id = location.hash.substring(1);
-		const container = await loadExistingContainer({
-			request: { url: id },
-			urlResolver,
-			documentServiceFactory,
-			codeLoader,
-		});
-		diceRoller = (await container.getEntryPoint()) as IDiceRoller;
+if (location.hash.length === 0) {
+	container = await createDetachedContainer({
+		codeDetails: { package: "1.0" },
+		urlResolver,
+		documentServiceFactory,
+		codeLoader,
+	});
+	await container.attach(createTinyliciousTestCreateNewRequest());
+	if (container.resolvedUrl === undefined) {
+		throw new Error("Resolved Url unexpectedly missing!");
 	}
-
-	render(diceRoller);
-	updateTabForId(id);
+	id = container.resolvedUrl.id;
+} else {
+	id = location.hash.substring(1);
+	container = await loadExistingContainer({
+		request: { url: id },
+		urlResolver,
+		documentServiceFactory,
+		codeLoader,
+	});
 }
 
-start().catch((error) => console.error(error));
```

Comment:
What was the idea behind removing `start()` and the `catch` block?

Reply:
Two main goals - one to just take advantage of having top-level await in mjs to provide a slightly simpler flow, and second to better match the main app with the test app which previously had slightly different calls/catches.